dd-sdk-android/src/main/kotlin/com/datadog/android/v2/core/internal/data/upload/DataUploadRunnable.kt

```diff
@@ -57,7 +63,9 @@ internal class DataUploadRunnable(
                     batch,
                     batchMeta
                 )
+                lock.countDown()
             }
+            lock.await(1, TimeUnit.SECONDS)
```

Comment:
actually it should be another value here, aligned with a connection timeout, I just forgot to update it :)

This lock is just introduced to deal with a *possible* async nature of the underlying storage, so that when we do `readNextBatch` call, we have 2 outcomes - no batch or there is a batch. So in each of these outcomes we unlock the lock, so that we can proceed. If callbacks are not called by some reason (for example, exception was thrown internally, but since that didn't lead to the crash), we will rely on this timeout to be able to proceed further instead of locking forever.